packages/api/src/command/medical/ihe-result/create-ihe-result.ts

```diff
@@ -37,21 +37,15 @@ export async function handleIHEResponse({ type, response }: IHEResult): Promise<
 
   switch (type) {
     case IHEResultType.PATIENT_DISCOVERY: {
-      const hasError = operationOutcome?.issue && !response.patientMatch;
-
-      await PatientDiscoveryResultModel.create({
-        ...defaultPayload,
-        status: hasError ? "success" : "failure",
-        data: response,
-      });
+      await createPatientDiscoveryResult(response);
       return;
     }
     case IHEResultType.DOCUMENT_QUERY: {
       const hasError = operationOutcome?.issue && !response.documentReference?.length;
 
       await DocumentQueryResultModel.create({
         ...defaultPayload,
-        status: hasError ? "success" : "failure",
+        status: hasError ? "failure" : "success",
```

Comment:
I noticed this when I merged develop. Think it was backwards, so I fixed it here

Reply:
NABD, but it would be a good function to avoid rewriting this multiple times